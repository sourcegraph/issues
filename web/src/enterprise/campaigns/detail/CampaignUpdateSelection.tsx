import * as H from 'history'
import * as React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { PageTitle } from '../../../components/PageTitle'
import { FilteredConnection } from '../../../components/FilteredConnection'
import { CampaignNode, CampaignNodeProps } from '../list/CampaignNode'
import { queryCampaigns } from '../global/list/backend'

interface Props {
    onSelect: (campaign: Pick<GQL.ICampaign, 'id'>) => void
    history: H.History
    location: H.Location
    className?: string
}

/**
 * A list of a campaign's changesets changed over a new plan
 */
export const CampaignUpdateSelection: React.FunctionComponent<Props> = ({ history, location, onSelect }) => (
    <>
        <PageTitle title="Update campaign" />
        <h1>Update campaign</h1>
        <p>Select the campaign to update from the list of campaigns below:</p>
        <FilteredConnection<
            Pick<
                GQL.ICampaign,
                'id' | 'closedAt' | 'name' | 'description' | 'changesets' | 'changesetPlans' | 'createdAt'
            >,
            Omit<CampaignNodeProps, 'node'>
        >
            history={history}
            location={location}
            nodeComponent={CampaignNode}
            nodeComponentProps={{
                selection: { buttonLabel: 'Select', enabled: true, onSelect },
            }}
            queryConnection={queryCampaigns}
            useURLQuery={false}
            hideSearch={true}
            noun="campaign"
            pluralNoun="campaigns"
        />
    </>
)
