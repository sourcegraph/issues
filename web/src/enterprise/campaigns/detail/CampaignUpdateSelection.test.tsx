import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { CampaignUpdateSelection } from './CampaignUpdateSelection'

describe('CampaignUpdateSelection', () => {
    test('renders', () => {
        const renderer = createRenderer()
        renderer.render(
            <CampaignUpdateSelection
                history={undefined as any}
                location={undefined as any}
                onSelect={() => undefined}
                className="abc"
            />
        )
        expect(renderer.getRenderOutput()).toMatchSnapshot()
    })
})
