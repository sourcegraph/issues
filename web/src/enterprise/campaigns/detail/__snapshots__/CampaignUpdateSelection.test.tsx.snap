// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignUpdateSelection renders 1`] = `
<React.Fragment>
  <PageTitle
    title="Update campaign"
  />
  <h1>
    Select campaign to update
  </h1>
  <p>
    Choose a campaign to update from the list below to preview which changes will be made to the code hosts:
  </p>
  <FilteredConnection
    defaultFirst={20}
    hideSearch={true}
    nodeComponent={[Function]}
    nodeComponentProps={
      Object {
        "selection": Object {
          "buttonLabel": "Preview",
          "enabled": true,
          "onSelect": [Function],
        },
      }
    }
    noun="campaign"
    pluralNoun="campaigns"
    queryConnection={[Function]}
    useURLQuery={false}
  />
</React.Fragment>
`;
